### DISCUSSION TITLE
Storage support and performance

### DISCUSSION THREAD

**Discussion by leo79901** [apache/cloudstack#9919] [Q&A] [answered]
Hi,

  I am new to Cloudstack and I have successfully installed and run VMs with NFS. Now I have a question and need help.
  To confirm the performance and stability, we are currently using storages such as Dell Unity 380 or Netapp AFF250 with FC-SAN ports. Through testing, I found that I can use these storages in three ways with Cloudstack:

  1. Use the NFS port from the storage.
  2. Use the storage as a local disk via SAN.
  3. Use the storage as a shared mount point with OCFS2 or GFS via SAN. (According to the official documentation, I guess, because I haven't achieved it.)

  The third option requires installing and maintaining a cluster file system, which is so difficult to use and maintain.
  The second option cannot enable high availability for VMs and is not suitable for business-critical applications.
  The first option is the most useful for me, but I am very cautious about its performance. Can it be used for a large database like PostgreSQL? 
​  Or which type do you usually use to support databases?

  Thanks a lot.

**Comment by alexandremattioli** [comment, marked as answer]
@leo79901  I've designed and ran one of the largest ACS deployments out there, most of the storage was NetApp with NFS and never really had any performance issues, been up for some 12 years now. It worked for 99% of customers. A few customers had dedicated storage and wanted iSCSI, so we just presented in a cloudstack network and connected directly to their VMs.

**Reply by leo79901** [reply]
Thanks a lot! Really helpful to me!
Yes, your answer has inspired me. For better performance, we can share a block to vm via iSCSI. Then write some scripts for VMs. With these scripts, the VMs boot, run the scripts, start the database, and run faster.
